src/test/java/org/cactoos/scalar/FirstOfTest.java

```diff
@@ -118,4 +118,28 @@ public void throwsFallbackIfNothingMatches() {
             new Throws<>(IllegalArgumentException.class)
         ).affirm();
     }
+
+    @Test
+    public void returnsFirstValueWithScalarFallback() {
+        new Assertion<>(
+            "Returns first value with scalar fallback",
+            new FirstOf<Integer>(
+                new IterableOfInts(2, 1, 0),
+                () -> -1
+            ),
+            new ScalarHasValue<>(2)
+        ).affirm();
+    }
+
+    @Test
+    public void returnsFirstValueWithIntegerFallback() {
+        new Assertion<>(
+            "Returns first value with Integer fallback",
+            new FirstOf<>(
+                new IterableOfInts(2, 1, 0),
+                -1
+            ),
+            new ScalarHasValue<>(2)
+        ).affirm();
```

Comment:
@iakunin can you just add one test for the empty iterable version?

Reply:
@victornoel sure. Done.